pkg/controllers/override/commandargspatch.go

```diff
@@ -0,0 +1,154 @@
+/*
+Copyright 2023 The KubeAdmiral Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package override
+
+import (
+	"encoding/json"
+	"fmt"
+	"sort"
+
+	corev1 "k8s.io/api/core/v1"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	fedcorev1a1 "github.com/kubewharf/kubeadmiral/pkg/apis/core/v1alpha1"
+	podutil "github.com/kubewharf/kubeadmiral/pkg/util/pod"
+)
+
+// parseCommandOrArgsOverriders parse OverridePatches from command/args overriders
+func parseCommandOrArgsOverriders(
+	fedObject fedcorev1a1.GenericFederatedObject,
+	overriders []fedcorev1a1.EntrypointOverrider,
+	target string,
+) (fedcorev1a1.OverridePatches, error) {
+	// patchMap(<targetPath, []target]>) is used to store the newest command/args values of each command path
+	patchMap := make(map[string][]string)
+
+	// parse patches and store the new command/args values in patchMap
+	for index := range overriders {
+		// get gvk from fedObj
+		gvk, err := getGVKFromFederatedObject(fedObject)
+		if err != nil {
+			return nil, err
+		}
+		podSpec, err := podutil.GetResourcePodSpec(fedObject, gvk)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get podSpec from sourceObj: %w", err)
+		}
```

Comment:
Can these lines be moved outside the loop?

Reply:
yes, the code has been updated.